libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1065,8 +1065,9 @@ void SMTEncoder::arrayIndexAssignment(Expression const& _expr, smtutil::Expressi
 		{
 			auto symbArray = dynamic_pointer_cast<smt::SymbolicArrayVariable>(m_context.expression(*base));
 			solAssert(symbArray, "");
+			auto baseType = base->annotation().type;
 			toStore = smtutil::Expression::tuple_constructor(
-				smtutil::Expression(base->annotation().type),
+				smtutil::Expression(make_shared<smtutil::SortSort>(smt::smtSort(*baseType)), baseType->toString(true)),
```

Comment:
Wait what, there were only 2 use cases for the separate constructor? :)

Reply:
Yea, it was a new thing, came with the new tuple stuff